diff --git a/dist/cjs/index-BuO47Yrm.js b/dist/cjs/index-BuO47Yrm.js
--- a/dist/cjs/index-BuO47Yrm.js
+++ b/dist/cjs/index-BuO47Yrm.js
@@ -707,7 +707,8 @@
          * The mount order and hierarchy is specific to ensure our element ref
          * is hydrated by the time features fire their effects.
          */
-        return (jsxRuntime.jsxs(MotionContext.Provider, { value: context, children: [MeasureLayout && context.visualElement ? (jsxRuntime.jsx(MeasureLayout, { visualElement: context.visualElement, ...configAndProps })) : null, useRender(Component, props, useMotionRef(visualState, context.visualElement, externalRef), visualState, isStatic, forwardMotionProps, isSVG, isValidProp)] }));
+        const rendered = useRender(Component, props, useMotionRef(visualState, context.visualElement, externalRef), visualState, isStatic, forwardMotionProps, isSVG, isValidProp);
+        return (jsxRuntime.jsxs(MotionContext.Provider, { value: context, children: [MeasureLayout && context.visualElement ? (jsxRuntime.jsx(MeasureLayout, { visualElement: context.visualElement, ...configAndProps }, "measure-layout")) : null, React.isValidElement(rendered) ? React.cloneElement(rendered, { key: "html" }) : rendered] }));
     }
     MotionDOMComponent.displayName = `motion.${typeof Component === "string"
         ? Component
diff --git a/dist/es/motion/index.mjs b/dist/es/motion/index.mjs
--- a/dist/es/motion/index.mjs
+++ b/dist/es/motion/index.mjs
@@ -1,7 +1,7 @@
 "use client";
 import { jsxs, jsx } from 'react/jsx-runtime';
 import { warning, invariant } from 'motion-utils';
-import { forwardRef, useContext } from 'react';
+import { cloneElement, forwardRef, isValidElement, useContext } from 'react';
 import { LayoutGroupContext } from '../context/LayoutGroupContext.mjs';
 import { LazyContext } from '../context/LazyContext.mjs';
 import { MotionConfigContext } from '../context/MotionConfigContext.mjs';
@@ -65,7 +65,8 @@
          * The mount order and hierarchy is specific to ensure our element ref
          * is hydrated by the time features fire their effects.
          */
-        return (jsxs(MotionContext.Provider, { value: context, children: [MeasureLayout && context.visualElement ? (jsx(MeasureLayout, { visualElement: context.visualElement, ...configAndProps })) : null, useRender(Component, props, useMotionRef(visualState, context.visualElement, externalRef), visualState, isStatic, forwardMotionProps, isSVG, isValidProp)] }));
+        const rendered = useRender(Component, props, useMotionRef(visualState, context.visualElement, externalRef), visualState, isStatic, forwardMotionProps, isSVG, isValidProp);
+        return (jsxs(MotionContext.Provider, { value: context, children: [MeasureLayout && context.visualElement ? (jsx(MeasureLayout, { visualElement: context.visualElement, ...configAndProps }, "measure-layout")) : null, isValidElement(rendered) ? cloneElement(rendered, { key: "html" }) : rendered] }));
     }
     MotionDOMComponent.displayName = `motion.${typeof Component === "string"
         ? Component
diff --git a/dist/framer-motion.dev.js b/dist/framer-motion.dev.js
--- a/dist/framer-motion.dev.js
+++ b/dist/framer-motion.dev.js
@@ -12522,7 +12522,8 @@
              * The mount order and hierarchy is specific to ensure our element ref
              * is hydrated by the time features fire their effects.
              */
-            return (jsxs(MotionContext.Provider, { value: context, children: [MeasureLayout && context.visualElement ? (jsx(MeasureLayout, { visualElement: context.visualElement, ...configAndProps })) : null, useRender(Component, props, useMotionRef(visualState, context.visualElement, externalRef), visualState, isStatic, forwardMotionProps, isSVG, isValidProp)] }));
+            const rendered = useRender(Component, props, useMotionRef(visualState, context.visualElement, externalRef), visualState, isStatic, forwardMotionProps, isSVG, isValidProp);
+            return (jsxs(MotionContext.Provider, { value: context, children: [MeasureLayout && context.visualElement ? (jsx(MeasureLayout, { visualElement: context.visualElement, ...configAndProps }, "measure-layout")) : null, React$1.isValidElement(rendered) ? React$1.cloneElement(rendered, { key: "html" }) : rendered] }));
         }
         MotionDOMComponent.displayName = `motion.${typeof Component === "string"
         ? Component
